.github/workflows/release.yaml

```diff
@@ -56,10 +56,140 @@ jobs:
         destination: tanzu-cli-admin-plugins
         credentials: ${{ secrets.GCP_BUCKET_SA }}
 
+    - name: Generate details about repo
+      id: generate-repo-details
+      shell: bash
+      run: |
+        echo "::set-output name=repo-name::$(echo '${{ github.repository }}' | awk -F '/' '{print $2}')"
+        echo "::set-output name=repo-org::$(echo '${{ github.repository }}' | awk -F '/' '{print $1}')"
+
+    - name: Get last release info
+      uses: octokit/request-action@v2.x
+      id: get-last-release
+      with:
+        route: GET /repos/{owner}/{repo}/releases/latest
```

Comment:
For last release: Getting the latest release will work only for minor version change. This will break for a patch release. I think the last release should be calculated based on the current release. For e.g if RELEASE_VERSION = X.Y.Z, LAST_RELEASE = X.Y.Z-1 if Z>0 X.Y-1.Z if Y>0 X-1.Y.Z if X>0. I think this is inline with semver versioning which we follow - https://semver.org/

Reply:
Updated to use semver based logic instead of api based retrieval. Only downside of this I believe is, we cannot skip any release version  eg- after 0.18.0 some reason we do not want 0.19.0 but 0.20.0 - not possible. However, using the latest release api has downside as well that would not work with patches. 
As a result, have modified to use semver based as suggested.